[7_1_X][TIMOB-24248] Android: Ti.Calendar Recurring Events are not clearly exposed

ypbnv opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24248

**Description:**
Cherry pick of: https://github.com/appcelerator/titanium_mobile/pull/9412

ypbnv committed:
Cherry pick of TIMOB-24248

build commented:
<!--
  0 failure: 
  0 warning: 
  1 messages
  
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-9869/3/artifact/dist/mobilesdk-7.1.0.v20180222043944-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

ypbnv committed:
Merge branch '7_1_X' into TIMOB-24248-7_1_X

ypbnv committed:
Always default to DAILY frequence.
Fix docs.
Add clang-format off for proxy methods.
Clear tresting code.
Fix MONTHLY rules.
Fix toStringArray() mothod.

lokeshchdhry commented:
FR Passed.

`Recurrence events` works as expected.

Studio Ver: 5.0.0.201712081732
SDK Ver: 7.1.0 local build
OS Ver: 10.13.2
Xcode Ver: Xcode 9.2
Appc NPM: 4.2.12
Appc CLI: 7.0.2
Daemon Ver: 1.0.1
Ti CLI Ver: 5.0.14
Alloy Ver: 1.11.0
Node Ver: 8.9.1
NPM Ver: 5.5.1
Java Ver: 1.8.0_101
Devices: ⇨ samsung SM-G955U1 --- Android 7.0
⇨ google Nexus 5 --- Android 6.0.1

Commit:
Merge branch '7_1_X' into TIMOB-24248-7_1_X
